docs/messageData.md

```diff
@@ -2939,8 +3065,11 @@ With this message the organizer lets the server know to expect a federation_init
     "public_key": "UvViTxoKsB3XVP_ctkmOKCJpMWb7fCzrcb1XDmhNe7Q=",
     "server_address": "wss://ethz.ch:9000/server",
     "challenge": {
```

Comment:
if the point here is to refer to the original challenge message, why not just refer to its message_id ? That would be entirely sufficient to someone having access to that message, seeing that it is a hash of the data and signature.

Reply:
We have to include the whole message here, because the 'original' challenge message (i.e. the one sent from the server to the organizer after a federation_challenge_request is not signed by the organizer). Upon receiving the message from the server, the organizer must extract the challenge data and create a new challenge message (now signed by him---the organizer) and pack that message into the federation_expect message s.t. the server can then again extract the challenge message from the federation_expect (which is now signed by the organizer) and send it back to the remote server as part of the federation result message.

Typing this out i realize this sounds very complicated but I think we discussed this and ultimately decided that  this is the best (and possibly only) way to implement the scheme with the trust boundaries wie defined.